superset-frontend/src/components/FilterableTable/FilterableTable.tsx

```diff
@@ -323,19 +323,19 @@ export default class FilterableTable extends PureComponent<
 
   sortResults(sortBy: string, descending: boolean) {
     return (a: Datum, b: Datum) => {
-      const aValue = a[sortBy];
-      const bValue = b[sortBy];
-      if (aValue === bValue) {
-        // equal items sort equally
-        return 0;
-      }
-      if (aValue === null) {
-        // nulls sort after anything else
-        return 1;
-      }
-      if (bValue === null) {
-        return -1;
-      }
+      // Parse any floating numbers so they'll sort correctly
+      const parseFloatingNums = (value: any) => {
+        const floatValue = parseFloat(value);
+        return Number.isNaN(floatValue) ? value : parseFloat(value);
+      };
+
+      const aValue = parseFloatingNums(a[sortBy]);
+      const bValue = parseFloatingNums(b[sortBy]);
+
+      String(aValue).localeCompare(String(bValue), undefined, {
```

Comment:
if you're using localeCompare with the numeric option, then do you still need to convert to a float? It looks like you're converting to a float and then back again to a string? It seems redundant.

Reply:
Also, nice find on localeCompare, but I don't think it's going to support multilingually unless you explicitly pass the language that you want to support as the second argument.

Reply:
I need to parse the floating numbers for them to sort correctly, but `localeCompare` will only take strings. The numerical option allows it to check the strings for numbers.

Reply:
@lyndsiWilliams but when you do `String(aValue)` and `String(bValue)` you're converting them back to strings. It should work if you just call `localeCompare` on the values without `parseFloat`, assuming you pass the `numeric` option.

Reply:
I did a lot of playing around with this and figured out that the sorting works without `localeCompare` so I removed it and added testing in [`this commit`](https://github.com/apache/superset/pull/17573/commits/8be6346d756a43203abf820c9e88d9c2f8f63b13). While testing I found any floating point number that had more than 2 floating points (like `12.001` or longer) is a `string`, while any other number is a `number`. I think originally, the floating numbers weren't sorting correctly because the types were mixed. Just using the `parseFloatingNums` function seems to solve this so maybe we can look at using `localeCompare` in the future for multilingual sorting, since it just seems to be causing issues in this instance. I also didn't realize you had to explicitly pass the language you want to support as a second argument, so I don't think I'll be able to make this work with `localeCompare`.